packages/camera/camera/example/ios/RunnerTests/StreamingTest.m

```diff
@@ -0,0 +1,85 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera;
+@import camera.Test;
+@import XCTest;
+@import AVFoundation;
+#import <OCMock/OCMock.h>
+#import "CameraTestUtils.h"
+
+@interface StreamingTests : XCTestCase
+@property(readonly, nonatomic) FLTCam *camera;
+@property(readonly, nonatomic) CMSampleBufferRef sampleBuffer;
+@end
+
+@implementation StreamingTests
+
+- (void)setUp {
+  dispatch_queue_t captureSessionQueue = dispatch_queue_create("testing", NULL);
+  _camera = FLTCreateCamWithCaptureSessionQueue(captureSessionQueue);
+  _sampleBuffer = FLTCreateTestSampleBuffer();
+}
+
+- (void)tearDown {
+  CFRelease(_sampleBuffer);
+}
+
+- (void)waitForStart {
+  NSPredicate *predicate = [NSPredicate predicateWithFormat:@"isStreamingImages == YES"];
+  XCTNSPredicateExpectation *expectation =
+      [[XCTNSPredicateExpectation alloc] initWithPredicate:predicate object:_camera];
+  XCTWaiterResult result = [XCTWaiter waitForExpectations:@[ expectation ] timeout:3];
+  XCTAssertEqual(result, XCTWaiterResultCompleted);
+}
+
+- (void)testExceedMaxStreamingPendingFramesCount {
+  XCTestExpectation *streamingExpectation = [self
+      expectationWithDescription:@"Must not call handler over maxStreamingPendingFramesCount"];
+
+  id handlerMock = OCMClassMock([FLTImageStreamHandler class]);
+  OCMStub([handlerMock eventSink]).andReturn(^(id event) {
+    [streamingExpectation fulfill];
+  });
+
+  id messenger = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
+  [_camera startImageStreamWithMessenger:messenger imageStreamHandler:handlerMock];
+
+  [self waitForStart];
```

Comment:
i highly recommend not to create this helper function since it's making the test a bit harder to read. Unit tests themselves are not unit tested, so it's better to make it as simple as possible.

Reply:
I see. Fixed.